sycl/source/detail/queue_impl.cpp

```diff
@@ -112,6 +112,7 @@ event queue_impl::memset(const std::shared_ptr<detail::queue_impl> &Self,
     MemoryManager::fill_usm(Ptr, Self, Count, Value,
                             getOrWaitEvents(DepEvents, MContext),
                             &EventImpl->getHandleRef(), EventImpl);
+    EventImpl->flushIfNeeded();
```

Comment:
It looks like after these changes we might start flushing queue after each usm operation even if all usm operations are submitted to the same queue, for example in a loop. Isn't it going to hurt performance? Please correct me if my understanding is wrong.

Reply:
I second @againull's concerns here. I think flushing after every operation is excessive. In the scheduler, we only perform flushing for cross-queue dependencies when enqueuing a new task, and I see that stays that way with this patch. So I would expect a similar approach here: only flushing cross-queue dependencies before a task that's dependent on them is enqueued instead of flushing immediately after each submission.

Reply:
That's fair enough :+1: . Maybe I can try to change this to flush every time a user does a et_info<sycl::info::event::command_execution_status>. I think that would solve the bug that [9944](https://github.com/intel/llvm/pull/9944) is meant to address and would also allow removing the hack from the OpenCL adapter.

Reply:
Why would we need to do that? The "submitted" status represents submission to the SYCL runtime, not the backend, so can't we just return "submitted" for that query if the underlying event is CL_QUEUED?

Reply:
I guess I misunderstood what "submitted" status represents in sycl. If that's the case, I guess we can just return "submitted" like you say. I don't see why that wouldn't work. Thanks for the help, I will update this PR with that change :+1: